src/Application/Model/Ticket.php

```diff
@@ -851,6 +851,42 @@ public function resetSource(Comment $comment = null) {
 			
 			return true;
 		}
+
+		public function reset($handle_id = null, $ticket_types = []) {
+			foreach ($ticket_types as $ticket_type) {
+				if (!ctype_alnum($ticket_type)) {
+					throw new ModelException("ticket_types can only contain alpha numeric characters.");
```

Comment:
I'm not sure whether this is the correct Exception.

Reply:
This should be fine.